sympy/solvers/ode/systems.py

```diff
@@ -0,0 +1,375 @@
+from sympy import (Matrix, Derivative, Symbol)
+from sympy.solvers.deutils import ode_order
+from sympy.core.numbers import I
+from sympy.core.relational import Eq
+from sympy.core.symbol import Dummy
+from sympy.functions import exp, im, cos, sin, re
+from sympy.functions.combinatorial.factorials import factorial
+from sympy.matrices import zeros
+from sympy.utilities import numbered_symbols, default_sort_key
+from sympy.utilities.iterables import uniq
+from sympy.simplify import simplify
+
+
+def _get_func_order(eqs, funcs):
+    order = dict()
+    func_dict = dict()
+
+    for func in funcs:
+        if not order.get(func, False):
+            max_order = -1
+            eq_no = -1
+            for i, eq in enumerate(eqs):
+                order_ = ode_order(eq, func)
+                if max_order < order_:
+                    max_order = order_
+                    eq_no = i
+            if eq_no in func_dict:
+                list_func = []
+                list_func.append(func_dict[eq_no])
+                list_func.append(func)
+                func_dict[eq_no] = list_func
```

Comment:
Is this equivalent?
```
func_dict[eq_no] = [func_dict[eq_no], func]
```

Reply:
Yes it is. This piece of code was taken from ```classify_sysode```. I will update this in ```systems.py``` as well as in ```ode.py```.

Reply:
Probably this whole function can be simplified.

Reply:
@oscarbenjamin Yes, the computation of ```func_dict``` here is meaningless and I will remove it.

Reply:
Also, I don't know if the ```if``` condition mentioned in this function ```if not order.get(func, False):``` is required?

Reply:
Can we just replace the whole function with this:
```
return {func: max(ode_order(eq, func) for eq in eqs) for func in funcs}
```

Reply:
@oscarbenjamin Yup we can do that. I will update the function to do the same.